### PR TITLE
Conda(-forge) updates to release script

### PR DESCRIPTION
**Pull request by scopatz**
This is an attempt to add conda-forge to the release script.  It also tries to generalize the release script to make it more portable to other projects.

### CODE REVIEW

File: release.xsh [hunk 1]
```diff
@@ -83,16 +118,102 @@ def pipify():
...
+    cd /tmp
+    url = "https://github.com/{0}/{1}/archive/{2}.tar.gz"
+    url = url.format(org, repo, target)
+    curl -L -O @(url)
+    sha, _ = $(sha256sum @('{}.tar.gz'.format(target))).split()
+    cd @(oldpwd)
+    return sha
+
+
+def feedstok_repos(ghuser):
```
Comment (gforsyth) [inline release.xsh]: should be `feedstock_repos`

File: release.xsh [hunk 2]
```diff
@@ -83,16 +118,102 @@ def pipify():
...
+    """Acquires a github token, writes a credentials file, and returns
+    the token.
+    """
+    password = ''
+    while not password:
+        password = getpass('GitHub Password for {0}: '.format(ghuser))
+    note = 'github3.py release.xsh ' + PROJECT + ' ' + socket.gethostname()
+    note_url = PROJECT_URL
+    scopes = ['user', 'repo']
+    auth = github3.authorize(ghuser, password, scopes, note, note_url)
```
Comment (gforsyth) [inline release.xsh]: this should probably support 2FA
http://github3py.readthedocs.io/en/master/examples/two_factor_auth.html

**Comment by gforsyth** [comment]
Hey @scopatz -- this looks good to me.  Although I suppose we won't know if it works until we try it, yeah?

File: release.xsh (hunk 2 above)
Comment (scopatz) [inline release.xsh]: Yeah, maybe that can be another PR after this.

File: release.xsh (hunk 1 above)
Comment (scopatz) [inline release.xsh]: Thanks fix is in!

**Comment by gforsyth** [comment]
Ok, looks good to me.  I'll merge once CI finishes.

**Comment by scopatz** [comment]
> Although I suppose we won't know if it works until we try it, yeah?

@gforsyth - yeah that is correct. I would expect some minor updates during the next release, which I am happy to do.

**Comment by gforsyth** [comment]
I'll add in the 2FA in sec, too

**Comment by gforsyth** [comment]
(so I can use it)

**Comment by scopatz** [comment]
Awesome :)

**Comment by gforsyth** [comment]
Blarg.  Impatient.  We don't test `release.xsh` anyway.  Merging now.

**Comment by gforsyth** [comment]
Thanks, @scopatz!

**Comment by scopatz** [comment]
> Blarg. Impatient. We don't test release.xsh anyway. Merging now.

Haha!
